Interface: move EditorSkeleton to interface package

vindl opened the pull request:
## Description
<!-- Please describe what you have changed or added -->

Moves EditorSkeleton component from `block-editor` to `interface` package.

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->

Smoke test the post, site, and widgets editor and verify that they still work as expected.

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

vindl committed:
Interface: move EditorSkeleton to interface package

Addison-Stavlo reviewed (commented):
This looks comprehensive, I gave the editors a smoke test and things still seem to be in good shape.

I'm really not sure what package this should live in, but interface seems reasonable.  

What is the underlying reason for this migration?  To move things usable by both block/site editors to a more neutral package?

vindl commented:
>I'm really not sure what package this should live in, but interface seems reasonable.
> What is the underlying reason for this migration? To move things usable by both block/site editors to a more neutral package?

Yes, and we didn't have the adequate package for that before so `block-editor` was used temporarily. Also @youknowriad requested it here https://github.com/WordPress/gutenberg/pull/20698#issuecomment-606470096.

Addison-Stavlo commented:
Makes sense to me.

Do you think we need to note anything in the interface package Readme about these?  They don't seem to have been listed in the block-editor one, but maybe it makes more sense as a more permanent home? (or maybe wait since they are both `__experimental*` ? )

(note: for anyone else reading by 'both' I am also referring to #21334)

vindl commented:
> (or maybe wait since they are both __experimental* ? )

This is likely going to be their permanent home so I'm not sure if they should still be prefixed as `__experimental`. I'll defer this to other reviewers. As for the readme - I'm not sure what to write, but happy to include something if you or other folks have ideas. :)

chrisvanpatten commented:
Might be worth changing the naming to be a bit more generic, so it’s not specifically named around “editor”? Also I think it might be expected that class names are updated to be prefixed with `interface-`?

youknowriad commented:
> Might be worth changing the naming to be a bit more generic, so it’s not specifically named around “editor”?

I agree with this but when we do that, we should also make the "region labels" editable maybe using a "labels" prop or something like that.

youknowriad commented:
I think (correct me if I'm wrong @jorgefilipecosta ) that the interface package is a "bundled" package that doesn't leak its API to Core. This means we can safely remove "__experimental" and we can still do breaking changes in the futur if needed.

vindl commented:
> Might be worth changing the naming to be a bit more generic, so it’s not specifically named around “editor”?

I'm fine with that. @chrisvanpatten @youknowriad any suggestions for the alternative name?

chrisvanpatten commented:
@vindl InterfaceSkeleton? Or just Skeleton?

It's also probably worth coming up with an alternate name for the "publish" region. I'm not sure exactly what that corresponds to in the UI — would it be the publish panel in the normal editor? Perhaps something like "Skeleton Action" region or something? Obviously if the region names are configurable then the consuming package can choose an appropriate name.

vindl commented:
Renamed to `InterfaceSkeleton` in 3b4dd233518c7e1b40e1cb1dcb5196d1cee946cd

vindl commented:
> I'm not sure exactly what that corresponds to in the UI — would it be the publish panel in the normal editor? Perhaps something like "Skeleton Action" region or something?

Yes, it does. I renamed it to `actions` for now.

> I agree with this but when we do that, we should also make the "region labels" editable maybe using a "labels" prop or something like that.

I added the option to pass the labels as props. I'm not sure if changing the region names themselves makes much sense given that their structure and nesting is hardcoded in the component.

youknowriad commented on packages/interface/src/components/interface-skeleton/index.js:
```diff
@@ -24,72 +24,84 @@ function useHTMLClass( className ) {
+		sidebar: __( 'Editor settings' ),
+		actions: __( 'Editor publish' ),
+		footer: __( 'Editor footer' ),
+	};
```
Do you think we should drop the "Editor" from the default labels?

youknowriad replied:
I also feel it works better for most UIs without "Editor" (like widgets)

vindl replied:
Yes, that sounds fine to me. 👍

vindl replied:
Updated in 2e1d036ae02a7f5eceed238059815bb05e750284.

youknowriad replied:
So this changes the labels for the widgets and more importantly the "Post editor". For me it's still good names and I like the fact that they are the same across pages. Let's keep them consistent for now and if we have feedback suggesting to go specific for some of these screens, we'll adapt.

afercia replied:
> Do you think we should drop the "Editor" from the default labels?

When used to label the ARIA landmark regions in the context of the WP admin, these labels need to differentiate the Editor landmarks from the _other_ landmarks in the WP UI. This is the reason why they used the word "Editor" at the beginning. For some history, see https://github.com/WordPress/gutenberg/pull/7940, https://github.com/WordPress/gutenberg/issues/7938, and https://github.com/WordPress/gutenberg/pull/2380

Also, the ARIA landmarks labels need to describe the purpose of the content within the region. Labels like "Header", "Footer", and "Left Sidebar" (added in https://github.com/WordPress/gutenberg/pull/20951) are arguably og any usefulness for assistive technology users.

If these labels are going to be used also for other purposes, then we'll need to provide specific labels for the ARIA landmarks.

Worth also noting the ARIA regions labels usage is still documented in the regions Readme:
> For better accessibility, these elements must be properly labelled to briefly describe the purpose of the content in the region. For more details, see "ARIA landmarks" in the [WAI-ARIA specification](https://www.w3.org/TR/wai-aria/).

Will create a new issue.

youknowriad commented on packages/edit-site/src/components/editor/index.js:
```diff
@@ -61,6 +61,14 @@ function Editor( { settings: _settings } ) {
 		};
 	}, [] );
 
+	const skeletonLabels = {
```
I wonder if we really need these now, as the default ones apply generically :)

vindl replied:
Pretty easy to change them later on if needed. I'm going to merge this to avoid more of the rebase hell. 😄

vindl replied:
Actually scratch that, I updated this in 68c5e2613d6722c435eb3eac112461335b772ecc after reading your comment [here](https://github.com/WordPress/gutenberg/pull/21335#discussion_r402963069).

vindl committed:
Remove custom site editor labels

jorgefilipecosta commented:
> I think (correct me if I'm wrong @jorgefilipecosta ) that the interface package is a "bundled" package that doesn't leak its API to Core.

That is correct :)
